### PR TITLE
fix: [CDS-58678]: Removed trim from identifier validation schema and updated Artifact and Manifest Identifier field to work as identifier

### PR DESCRIPTION
**Pull request by pankajsingh-harness**
### Summary

- Removed trim() from the IdentifierSchemaWithOutName as when user was adding leading and trailing spaces, the validations still pass but it comes as `identifier: "     test_test   "` in YAML and BE send the error: `message
: 
"Invalid yaml: $.service.serviceDefinition.spec.artifacts.primary.sources[0].identifier: does not match the regex pattern ^[a-zA-Z_][0-9a-zA-Z_]{0,127}$"`. This will will make UI and BE validation same.

- Updated Artifact and Manifest Identifier label and make that work as Id field in InputWithIdentifier.

**Note:** There any multiple snapshot update due to uicore version update [especially this change](https://github.com/harness/uicore/pull/1022). These changes in snapshots are placeholder deletion from the layout component (`Layout.Horizontal`) which does not have any usecase. As having placeholder for layout component(`div`) does not have any visible change. 

UI Core PR: [https://github.com/harness/uicore/pull/1026/files](https://github.com/harness/uicore/pull/1026/files)

#### Screenshots

Screenshot 2023-04-06 at 11 18 56 AM

Screenshot 2023-04-06 at 11 19 31 AM

Screenshot 2023-04-06 at 11 24 49 AM

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/10-common/components/Identifier/Identifier.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { useFormikContext } from 'formik'
+import type { FormikContextProps } from '@harness/uicore/dist/components/FormikForm/utils'
+import { getIdentifierFromName } from '@common/utils/StringUtils'
+
+function Identifier(props: Omit<TextProps & FormikContextProps<any>, 'onChange'>): React.ReactElement {
+  const { setFieldValue } = useFormikContext()
+
+  return (
+    <FormInput.Text
+      onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (AshwinBhatkal-Harness) [inline src/modules/10-common/components/Identifier/Identifier.tsx:20]: The field value should change irrespective of this onChange. Why the specific one?

Comment (pankajsingh-harness) [reply src/modules/10-common/components/Identifier/Identifier.tsx:20]: @AshwinBhatkal-Harness this is the identifier field. This implementation is same as identifier component in uicore https://github.com/harness/uicore/blob/main/packages/uicore/src/components/InputWithIdentifier/InputWithIdentifier.tsx#L122
I have created this re-usable component to be used where we just need identifier field without name. For the artifact and manifetst, UI was showing this field as Name but it actually identifier field, so all the issues of validation. I have updated this at all places,

**Comment by pankajsingh-harness** [comment]
retrigger cypress-cv

File: src/modules/10-common/components/Identifier/Identifier.tsx (hunk 1 above)
Comment (pankajsingh-harness) [reply src/modules/10-common/components/Identifier/Identifier.tsx]: @AshwinBhatkal-Harness Removed this new component and updated uicore `FormInput.Text` component to support id field behaviour.

**Review by chetanvekariya-harness — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -63,7 +63,7 @@
     "@harness/monaco-yaml": "^1.0.0",
     "@harness/ng-tooltip": "^1.31.101",
     "@harness/telemetry": "^1.0.42",
-    "@harness/uicore": "3.121.0",
+    "@harness/uicore": "3.125.0",
```
Comment (YounixIO) [inline package.json:66]: From 121 to 125 ?

**Review by YounixIO — APPROVED** [review]

**Review by sonaligoyal-harness — APPROVED** [review]

**Pull request merged**